### ISSUE TITLE
I think I got native nvram to work on Z390

### ISSUE THREAD

**Issue by hylkefaber** [khronokernel/Opencore-Vanilla-Desktop-Guide#17]
So I was experimenting with OpenCore on my new Z390 build (Gigabyte Z390 Designare, 9900KS, Vega64) and trying to get emulated NVRAM to work with FwRuntimeServices.efi on Clover. Sadly, this didn't work: new nvram variables didn't survive a reboot. After trying various different OpenCore settings, I finally managed to boot into Catalina with OC by disabling my iGPU and using SMBIOS iMacPro1,1. Just for fun I downloaded and booted with a DSDT I found on the Z390 Designare thread on olarila.com and lo and behold, my nvram variables survived!
I was always under the impression that native nvram was a no-go, due to the information in the buying guides and the Anti-Hackintosh Buying Guide in the sidebar, but now I think this might be due for an update.
Here's the thread (https://olarila.com/forum/viewtopic.php?p=91119#p91119) where I found my DSDT, you can even ask the guy to make a custom DSDT specifically for your setup but you have to download and run an app called RunMe.app and I'm not sure exactly what that does. Because I was just experimenting I made sure I wasn't connected to the internet when I ran the app and fully reinstalled Catalina afterwards, just to be sure. Also, the guy responds faster if you donate via PayPal first, but I guess that is only fair? I'm not sure what edits he makes, sadly he seems unwilling to share his DSDT patches.
I tried to mention this in the big CaseySJ Z390 Designare thread over on tonymacx86 but you get banned for mentioning olarila.com, the Vanilla guide or even OpenCore.... Even my post on Reddit got banned, but this one (https://www.reddit.com/r/hackintosh/comments/ed6f0n/i_think_i_might_have_gotten_native_nvram_to_work/) survived.
tl;dr I might be wrong but I think I have gotten native nvram to work on Z390 by using a DSDT

**Comment by khronokernel** [comment]
Sorry but this was likely a one in a million chance of working NVRAM as we know that the NVRAM variables are read only with the true 300 series boards and go into lock when you try to write. A DSDT patch wouldn't bring back the function of NVRAM write though I'd be willing to look into it further if you'd be willing to provide the files. What I'd like is the original DSDT and the new one you're using, I can see what kind of patches were made and determine from there if they'd affect NVRAM. I have my doubts but wouldn't mind looking into it further

**Comment by hylkefaber** [comment]
Sure I would love to help! I've posted a number of threads on Reddit but they got deleted because I mentioned some sources that they didn't like, but I think I'm free to describe my process here. So I have uploaded my clean DSDT and the modified DSDT here (http://s000.tinyupload.com/index.php?file_id=40561690984440084401) The file was patched by Maldon on olarila.com, but I wasn't able to figure out which patches were used. There's a list describing them [here](https://olarila.com/forum/viewtopic.php?t=11237), but again, I have no idea what exactly happened. We can only compare the original and the result.
Here's the EFI I used for booting the system (http://s000.tinyupload.com/index.php?file_id=38748510796799477343). I hope I can bring something useful to the table! I'm running very stable for about two weeks now.

**Comment by khronokernel** [comment]
Sorry but these DSDT patches still wouldn't affect NVRAM, I've met one other person with working NVRAM on Z390 Designare so it seems to be a hit or miss depending on the user. With the chances being quite rare and that user having odd NVRAM quirks.

I apologies but going to have to close this issue as no sustainable info can be pulled from it but thank you for providing the files s quickly.

**Comment by hylkefaber** [comment]
I’m not sure I understand it myself. Over on the Tonymacx86 CaseySJ Z390 Designare thread there’s a couple of users that have gotten it to work. I know it sounds ridiculous to suggest that a DSDT can fix nvram, but when I boot without this DSDT and use the native one, my nvram variables don’t survive a reboot, they only do when booting with this file. I hope this finding can help some users out.

**Comment by KvL-KvL** [comment]
Hello, I might be wrong, but the up-to-date BIOS (Rev. 10c) on my Gigabyte Z390 Gaming X booted with OpenCore 0.5.5 and slide=0 seems to work perfectly with native NVRAM. I followed the Vanilla Giuide [here](https://khronokernel-2.gitbook.io/opencore-vanilla-desktop-guide/).
Variables put with `sudo nvram MyVar=Test` is correctly retained across reboots, even ctrl+enter boot to select the preferret boot disk in OpenCore works.
K.

**Comment by dreamwhite** [comment]
> So I was experimenting with OpenCore on my new Z390 build (Gigabyte Z390 Designare, 9900KS, Vega64) and trying to get emulated NVRAM to work with FwRuntimeServices.efi on Clover. Sadly, this didn't work: new nvram variables didn't survive a reboot. After trying various different OpenCore settings, I finally managed to boot into Catalina with OC by disabling my iGPU and using SMBIOS iMacPro1,1. Just for fun I downloaded and booted with a DSDT I found on the Z390 Designare thread on olarila.com and lo and behold, my nvram variables survived!
> I was always under the impression that native nvram was a no-go, due to the information in the buying guides and the Anti-Hackintosh Buying Guide in the sidebar, but now I think this might be due for an update.
> Here's the thread (https://olarila.com/forum/viewtopic.php?p=91119#p91119) where I found my DSDT, you can even ask the guy to make a custom DSDT specifically for your setup but you have to download and run an app called RunMe.app and I'm not sure exactly what that does. Because I was just experimenting I made sure I wasn't connected to the internet when I ran the app and fully reinstalled Catalina afterwards, just to be sure. Also, the guy responds faster if you donate via PayPal first, but I guess that is only fair? I'm not sure what edits he makes, sadly he seems unwilling to share his DSDT patches.
> I tried to mention this in the big CaseySJ Z390 Designare thread over on tonymacx86 but you get banned for mentioning olarila.com, the Vanilla guide or even OpenCore.... Even my post on Reddit got banned, but this one (https://www.reddit.com/r/hackintosh/comments/ed6f0n/i_think_i_might_have_gotten_native_nvram_to_work/) survived.
> tl;dr I might be wrong but I think I have gotten native nvram to work on Z390 by using a DSDT

@hylkefaber please note that if you're using clover, in order to use also FWRuntimeServices you need to add OCQuirks (https://github.com/ReddestDream/OcQuirks).
